Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py

```diff
@@ -285,116 +288,129 @@ def install(self):
         for dependency in self.implicit_deps:
             AutoInstall.install(dependency)
 
-        if not self.archives():
-            raise ValueError('No archives for {}-{} found'.format(self.pypi_name, self.version))
-        archive = self.archives()[-1]
-
-        try:
-            install_location = AutoInstall.directory
-            shutil.rmtree(self.location, ignore_errors=True)
-
-            AutoInstall.log('Downloading {}...'.format(archive))
-            archive.download()
-
-            temp_location = os.path.join(tempfile.gettempdir(), '{}-{}'.format(self.name, os.getpid()))
-            archive.unpack(temp_location)
-
-            for candidate in os.listdir(temp_location):
-                candidate = os.path.join(temp_location, candidate)
-                if not os.path.exists(os.path.join(candidate, 'setup.py')):
-                    # If a package has a setup.cfg, we can just lay down a dummy setup.py and let setuptools handle the rest
-                    if os.path.exists(os.path.join(candidate, 'setup.cfg')):
-                        with open(os.path.join(candidate, 'setup.py'), 'w') as setup_py:
-                            setup_py.write('from setuptools import setup\n')
-                            setup_py.write('if __name__ == "__main__":\n')
-                            setup_py.write('    setup()\n')
-                    else:
-                        continue
+        lockfile = os.path.join(AutoInstall.directory, AutoInstall.LOCK_FILE)
+        if os.path.exists(lockfile) and time.time() - os.path.getmtime(lockfile) > AutoInstall.LOCKFILE_TIMEOUT:
+            os.unlink(lockfile)
 
-                AutoInstall.log('Installing {}...'.format(archive))
+        with FileLock(lockfile, timeout=AutoInstall.LOCKFILE_TIMEOUT):
+            # Re-load our manifest, in case another process is using this autoinstall location
```

Comment:
Will clarify this comment, yes, it's because another process could have (and, in fact, is likely to have) installed the package we're about to install.